### PR TITLE
Refactored message-list, made ref work again

### PR DESCRIPTION
**Pull request by larstadema**

### CODE REVIEW

File: src/message-list/index.jsx [hunk 1]
```diff
@@ -3,107 +3,96 @@ import PropTypes from 'prop-types';
...
+const propTypes = {
     /** MessageList content */
     children: PropTypes.node.isRequired,
     /** The amount of pixels the user has to scroll up to disable auto scroll */
     scrollOffset: PropTypes.number,
-    /** Expose the components methods to the parent,
-     * useful for calling scrollDown from a parent component */
-    addRef: PropTypes.func,
+    /** Reference list element */
+    listRef: PropTypes.func,
```
Comment (sjaakluthart) [inline src/message-list/index.jsx]: Why did you change this prop's name? Changing this means breaking the current API.

File: src/message-list/index.jsx [hunk 2]
```diff
@@ -3,107 +3,96 @@ import PropTypes from 'prop-types';
...
+
+    if (autoScroll && this.shouldScroll) {
+      this.scrollToBottom();
+
+      this.shouldScroll = false;
+    }
   }
 
-  shouldScrollToBottom(node) {
+  shouldScrollTo = ({ scrollHeight, scrollTop, offsetHeight }) => {
```
Comment (sjaakluthart) [inline src/message-list/index.jsx]: This naming is a little weird, the list will always scroll to the bottom, right?

File: src/message-list/index.jsx (hunk 1 above)
Comment (larstadema) [reply src/message-list/index.jsx]: Because the naming is wrong, and not like all the other ref namings

Comment (sjaakluthart) [reply src/message-list/index.jsx]: Ok, then this PR should be part of the 5.0.0 release!

Comment (larstadema) [reply src/message-list/index.jsx]: Calling prototype functions from this component, should just be done by the parent setting a ref. The correct ref passthrough is done from the ul element to the parent.

Comment (IanCStewart) [reply src/message-list/index.jsx]: I have changed the base branch to `v5.0.0`

Comment (sjaakluthart) [reply src/message-list/index.jsx]: I'll approve it, can you fix the linting errors Travis mentioned?

Comment (larstadema) [reply src/message-list/index.jsx]: An example usage of calling scrollToBottom: 

```
parentScrollToBottom = () => {
  this.node.scrollToBottom();
}

render() {
  <MessageList ref={(node) => { this.node = node; }} />
}
```
